extra_geom/base.py

```diff
@@ -791,3 +798,135 @@ def offset(self, shift, *, modules=np.s_[:], tiles=np.s_[:]):
                 for t, tile in enumerate(module)
             ] for m, module in enumerate(self.modules)
         ])
+
+    def rotate(self, angles, center=None, modules=np.s_[:], tiles=np.s_[:],
+               degrees=True):
+        """Rotate part or all of the detector, making a new geometry.
+
+        By default, this rotates all modules & tiles.
+        Returns a new geometry object of the same type.
+
+        ::
+
+            # Rotate the whole geometry by 90 degree in the xy plan
+            geom2 = geom.rotate((0, 0, 90))
+
+            # Move the tile 0 in the module 0 around its center by 90 degrees
+            geom2 = geom.rotate((0, 0, 90), modules=np.s_[:1], tiles=np.s_[:1])
+
+            # Rotate each module by a separate amount
+            rotate = np.zeros((16, 3))
+            rotate[5] = (3, 5, 1)  # x, y, z for individual modules
+            rotate[10] = (0, -2, 1)
+            geom2 = geom.rotate(rotate)
```

Comment:
I'd like to have a brief description of what a rotation in each dimension means - maybe this is obvious to other people, but it always hurts my head that you specify an angle for z to rotate in the x-y plane (I understand why, it's just not intuitive).

In particular, I think it would be useful to have a reminder which way is 'forwards' for each dimension - e.g. is z rotation clockwise looking along the beam, or anticlockwise? Again, maybe this is second nature to some users, but probably not all.

Reply:
I tried my best, let me know if that's clear enough 🙃